File: 86.R
# Cuboid route

Args  <- commandArgs()
target <- ifelse(is.na(Args[6]), 1000000, as.numeric(Args[6]))

total <- 0
m <- 1
while ( total < target ) {
    for ( ij in seq.int(2 * m) ) {
        route = sqrt(ij * ij + m * m)
        if ( as.integer(route) == route ) {
            if ( ij <= m ) {
                total <- total + ij %/% 2
            } else {
                total <- total + m - (ij + 1) %/% 2 + 1
            }
        }
    }
    m <- m + 1
}

cat(m - 1, fill=TRUE)

File: 86.js
#!/usr/bin/env node

// Cuboid route

var program = require('commander');

program
    .version('0.1.0')
    .description('Cuboid route')
    .option('-t, --target <int>',
        'The target number of distinct cuboids', Number, 1000000)
    .parse(process.argv);

var total = 0;
var m = 1;
while (total < program.target) {
    for (var i = 1; i <= m; i++) {
        for (var j = i; j <= m; j++) {
            var route = Math.sqrt((i + j) * (i + j) + m * m);
            if (Math.floor(route) == route) {
                total++;
            }
        }
    }
    m++;
}

console.log(m - 1);
